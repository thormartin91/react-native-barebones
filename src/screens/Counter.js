/** @flow */

'use strict'

import React from 'react';
import { StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';
import type { Action } from '../actions/types';
import { increment, decrement } from '../actions/counter';
import { Colors } from '../components/Colors';
import Text from '../components/Text';
import Header from '../components/Header';
import Button from '../components/Button';

type Props = {
  state: {
    count: number
  },
  increment: () => Action,
  decrement: () => Action
};

class CounterScreen extends React.Component {

  props: Props;

  render() {
    return (
      <View style={styles.container}>
        <Header>{ this.props.state.count }</Header>
        <Button onPress={ this.props.increment } style={{borderRadius: 50}}>
          <Text>up</Text>
        </Button>
        <Button onPress={ this.props.decrement } caption={"down"}>
        </Button>
      </View>
    );
  }
  
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.lightBlue,
  }
});


const mapStateToProps = (state) => {
  return {
    state: state.counter
  };
};

const mapDispatchToProps = {
  increment,
  decrement
};


export default connect(mapStateToProps, mapDispatchToProps)(CounterScreen);
