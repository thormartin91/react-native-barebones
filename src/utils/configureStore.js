/** @flow */

'use strict';

import {applyMiddleware, createStore} from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';


export default function configureStore() {
  var createAppStore = applyMiddleware(thunk)(createStore);
  const store = createAppStore(rootReducer)
  return store;
}
