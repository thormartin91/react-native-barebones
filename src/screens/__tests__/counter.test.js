'use strict'

import 'react-native';
import React from 'react';
import Counter from '../Counter';
import renderer from 'react-test-renderer';
import configureStore from '../../utils/configureStore';

const store = configureStore();

describe('Counter-screen', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
        <Counter store={store}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
