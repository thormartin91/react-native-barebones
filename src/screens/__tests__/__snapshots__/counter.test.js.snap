exports[`Counter-screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgba(32,32,32,1)",
          "fontSize": 25,
          "margin": 10,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    0
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderRadius": 50,
        "height": 30,
        "justifyContent": "center",
        "margin": 3,
        "opacity": 1,
        "padding": 10,
        "width": 100,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(32,32,32,1)",
          },
          undefined,
        ]
      }>
      up
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "height": 30,
        "justifyContent": "center",
        "margin": 3,
        "opacity": 1,
        "padding": 10,
        "width": 100,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(32,32,32,1)",
          },
          Object {
            "color": "rgba(32,32,32,1)",
          },
        ]
      }>
      down
    </Text>
  </View>
</View>
`;
